server/platform/services/sharedchannel/sync_recv.go

```diff
@@ -65,6 +65,7 @@ func (scs *Service) processSyncMessage(c request.CTX, syncMsg *model.SyncMsg, rc
 		mlog.Int("user_count", len(syncMsg.Users)),
 		mlog.Int("post_count", len(syncMsg.Posts)),
 		mlog.Int("reaction_count", len(syncMsg.Reactions)),
+		mlog.Int("status_count", len(syncMsg.Statuses)),
```

Comment:
Didn't add errors to the `model.SyncResponse` right above in this first pass as currently erroring when saving statuses in the app layer logs a WARN instead of returning and error

Reply:
Shouldn't we add the field here, and handle the errors on the sender side so that once server is changed to provide those errors, slightly older servers will handle them correctly?

Reply:
Added error management on https://github.com/mattermost/mattermost/pull/28020/commits/81a7ee2099af50cd966eca4ff059a88e7e31d663, with a similar approach to the one we use for posts. I'm not sending back and forth the status contents but fetching the new status for the retry message to avoid race conditions

PTAL

Reply:
Perfect.  Looks great.